internal/util/update/resource-merge.go

```diff
@@ -124,6 +126,10 @@ func (u ResourceMergeUpdater) updatePackage(subPkgPath, localPath, updatedPath,
 		// we don't re-add the updated package from upstream.
 	// Package deleted from upstream
 	case originalExists && localExists && !updatedExists:
+		// make sure that merge comments are added before comparing to avoid unwanted diffs
+		if err := addmergecomment.Process(localPath, originalPath); err != nil {
```

Comment:
I don't follow why we are adding the merge comments at these locations in the code. It seems like we should decide on a strategy for when we should add these comments, and then based on that decide where in the codebase we should make the call.

So I know there are at least two cases:
 * If we pull in new resources in the local fork that doesn't already have the merge comment, we want to add it.
 * Sometimes we want to decide if there are local changes to a package and we don't want any merge comments added in the local fork to be interpreted as local changes to a fork that would prevent deletion. 
I'm not sure this is complete.

Based on this, it seems like we should add any necessary merge comments after a package has been updated/fetched, regardless of how it happened (merge strategy, new subpackage added...).
For the diff situation, it seems like we should either make these changes consistently for every package during updates, or handle it within the diff functionality. This seems like we update the packages with this information in just some situations.

Reply:
Agree. These should be added in the update.go. Before we invoke merge operation, we should get all the three sources on level ground by adding merge comments to all of them so that the update would work just fine like today. 

So in update workflow these are invoked twice:
1. After doing pre-checks and just before invoking updaters of individual strategies.
2. At the end, to make sure that the all updated resources on local have the merge comments as a final ensuring step.

Individual strategies should not have to worry about this.